server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -124,9 +126,17 @@ public void onAfterTranslogSync() {
                 engineConfig.getPrimaryModeSupplier()
             );
             this.translogManager = translogManagerRef;
-        } catch (IOException e) {
-            IOUtils.closeWhileHandlingException(store::decRef, readerManager, translogManagerRef);
+            success = true;
```

Comment:
Something similar could happen in the catch block too, but it would likely be flagged as unreachable code by the analyzers and is more likely to be caught and fixed.

Reply:
This is similar to the handling in `InternalEngine` as well .  Since the `decRef` is in `finally` block, it shouldn't get skipped .

Reply:
@gbbafna even though the decRef is in the `finally` block, it is wrapped inside the `success == false` check